Introductory Tests/udp_initial_condition.py

```diff
@@ -145,6 +145,23 @@ def euler_approx(self, r, v, time_list, r_store, v_store, a_store):
         return r_store, v_store, a_store
 
 
+    def comet_collision_detection(self, x):
+        """
+        Uses is_outside to check if a set of positions (or current) x is is inside mesh.
+        Returns boolean with corresponding results.
+
+        Args:
+            x: Array containing current, or a set of, positions expressed in 3 dimensions.
+
+        Returns:
+            collision_boolean: A one dimensional array with boolean values corresponding to each
+                               position kept in x. Returns "False" if point is inside mesh, and 
+                               "True" if point is outside mesh (that is, there no collision).
+        """
+        collision_boolean = mesh_utility.is_outside(x, self.mesh_vertices, self.mesh_faces)
```

Comment:
Running möller trumbore is pretty costly. So before you check with it I would precompute somewhere a bounding box of the mesh and check if it is outside that bounding box. hopefully, in many cases it will be and save you the computational time.

Reply:
As the least computationally costly alternative, I tested this with a bounding sphere during the weekend which seemed to work fine. However, given the very non-convex shape of the comet, both an up-scaled bounding box, and even more so a bounding sphere, will generate large differences in distances from the comets surface to the boundary of the "collision-risk zone". 

As an alternative, which would likely also be more accurate, one idea could be to implement something lika a low-polytope convex hull around the mesh? Or do you think that it would be better to instead go for something like a low-low polytope but upscaled mesh of the comet right away (like you mentioned last Friday)?

Reply:
I would start with the sphere / bounding box. Performance profiling will show whether that leads to a too large performance impact? heyoka may also be an alternative, to be discussed tmr :v: